### ISSUE TITLE
[ISSUE] Unable to dismiss VGSCardIOScanController

### ISSUE THREAD

**Issue by thinhv** [verygoodsecurity/vgs-collect-ios#147]
**Describe the bug**
Not sure if I missed something but I can't dismiss the VGSCardIOScanController.

**To Reproduce**
1. Initialize `VGSCardIOScanController(cardScanDelegate)`
2. Call scanViewController.presentCardScanner(on: self, animated: true, completion: nil)
3. Press cancel nothing happen / after finish scanning delegate's method doesn't get called

**Expected behavior**
The scan view controller should be dismissed once user finishes scanning or user presses cancel button. (And delegate methods should be called)

**Environment (please complete the following information):**
 - SDK Version [e.g. 1.5.1]
 - iOS Version: [e.g. iOS 13.4.1]
 - Xcode Version: [e.g. Xcode 11.5.1]
 - Device: [e.g. iPhone X]
 - Dependency manager and it's version:
    - [ ] Cocoapods: [e.g.  1.9.1]

**Comment by dmytrokhl** [comment]
@thinhv here the working [example](https://github.com/verygoodsecurity/vgs-collect-ios/blob/master/demoapp/demoapp/ViewController.swift#L221). You should handle **cancel** delegate func like: 
```
func userDidCancelScan() {
   scanViewController.dismissCardScanner(animated: true, completion: nil)
}
```
Also check that your `scanViewController` instance still exists and stored on `ViewController` level(not func level).

Let me know if that works for you.

**Comment by isimpson** [comment]
@dmytrokhl it will work correctly when setting the `delegate` explicitly as in your example but there is an issue here:
```
    public required init(_ delegate: VGSCardIOScanControllerDelegate? = nil) {
        #if canImport(CardIO)
            self.scanHandler = VGSCardIOHandler()
            self.delegate = delegate
        #else
            print("Can't import CardIO. Please check that CardIO submodule is installed")
            return
        #endif
    }
```

If the delegate is passed into the initializer then it will never be set in the `scanHandler` here:
```
    public var delegate: VGSCardIOScanControllerDelegate? {
        didSet {
            scanHandler?.delegate = delegate
        }
    }
```

`didSet` isn't called in this case, see the swift docs https://docs.swift.org/swift-book/LanguageGuide/Properties.html

> NOTE
> The willSet and didSet observers of superclass properties are called when a property is set in a subclass initializer, after the superclass initializer has been called. They are not called while a class is setting its own properties, before the superclass initializer has been called.
> 

I suggest either you set the `scanHandler.delegate` explicitly in the init or remove the option to pass the delegate to the init to avoid confusion.

**Comment by dmytrokhl** [comment]
Thank you for your findings @isimpson @thinhv . The fix released in `VGSCollectSDK` [v1.5.2](https://github.com/verygoodsecurity/vgs-collect-ios/releases/tag/1.5.2).
